standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -3501,6 +3541,10 @@ public void addDynamicPartitions(AddDynamicPartitions rqst)
             pstmt.executeBatch();
           }
         }
+        try (PreparedStatement pstmt = dbConn.prepareStatement(TXN_COMPONENTS_DP_DELETE_QUERY)) {
```

Comment:
This fix is not related to the original problem, shouldn't it be committed separately?

Reply:
It's actually related. Operation that involve dynamic partitioning doesn't generate TXN_COMPONENTS at the locking stage, making it impossible to check for write conflict.